### PR TITLE
Fix: Tracing broken in GCE (app engine flexible) - Trivial

### PR DESCRIPTION
**Pull request by dhendry**
This PR fixes an issue we had when upgrading to opencensus-python 1.8 where tracing was totally broken. After some digging, I found this error:

```
StackdriverExporter failed to emit spans.Dropping 2 spans from queue.
Traceback (most recent call last):
  File "/env/lib/python3.6/site-packages/opencensus/trace/exporters/transports/background_thread.py", line 106, in _thread_main
    self.exporter.emit(span_datas)
  File "/env/lib/python3.6/site-packages/opencensus/trace/exporters/stackdriver_exporter.py", line 224, in emit
    stackdriver_spans = self.translate_to_stackdriver(trace)
  File "/env/lib/python3.6/site-packages/opencensus/trace/exporters/stackdriver_exporter.py", line 248, in translate_to_stackdriver
    set_attributes(trace)
  File "/env/lib/python3.6/site-packages/opencensus/trace/exporters/stackdriver_exporter.py", line 76, in set_attributes
    set_monitored_resource_attributes(span)
  File "/env/lib/python3.6/site-packages/opencensus/trace/exporters/stackdriver_exporter.py", line 105, in set_monitored_resource_attributes
    'project_id')
  File "/env/lib/python3.6/site-packages/opencensus/trace/exporters/stackdriver_exporter.py", line 136, in set_attribute_label
    label_value_prefix + resource_labels[attribute_key]
TypeError: must be str, not bytes"   
```

Basically `urlopen(request, timeout=_REQUEST_TIMEOUT).read()` returns bytes not a string.

This PR also fixes an annoying issue I highlighted in the last paragraph of https://github.com/census-instrumentation/opencensus-python/pull/203#issue-193674334

### CODE REVIEW

**Comment by liyanhui1228** [comment]
Thanks for this fix! I think #367 can also be fixed by this.

File: opencensus/common/monitored_resource_util/gcp_metadata_config.py [hunk 1]
```diff
@@ -95,6 +95,9 @@ def _initialize_metadata_service(cls):
         if instance_id is not None:
             cls.is_running = True
 
+            if isinstance(instance_id, bytes):
```
Comment (liyanhui1228) [inline opencensus/common/monitored_resource_util/gcp_metadata_config.py]: What about moving the type conversion logic to the function `_get_attribute` below?

File: opencensus/common/monitored_resource_util/gcp_metadata_config.py [hunk 2]
```diff
@@ -106,6 +109,12 @@ def _initialize_metadata_service(cls):
                 if attribute_key not in gcp_metadata_map:
                     attribute_value = cls._get_attribute(attribute_key)
                     if attribute_value is not None:
+                        if isinstance(attribute_value, bytes):
```
Comment (liyanhui1228) [inline opencensus/common/monitored_resource_util/gcp_metadata_config.py]: Same here.

**Comment by dhendry** [comment]
@liyanhui1228 Comment addressed, all tests passing, and merged from master. Let me know if there is anything else which needs to be done to get this PR in. It fixes some a problem which makes this library totally non-functional in our environment so an official release ASAP would be much appreciated.

**Comment by dhendry** [comment]
@liyanhui1228 Anything else blocking this? Its a trivial change. All comments addressed, tests passing, and latest master has been merged

**Comment by reyang** [comment]
@dhendry your change looks good to me. I can merge this if there is no objection (let's wait for a day to see if @liyanhui1228 and @c24t have comments).

**Comment by dhendry** [comment]
@reyang Awesome, thanks

**Pull request merged**